Neos.RedirectHandler.DatabaseStorage/Classes/RedirectionStorage.php

```diff
@@ -0,0 +1,240 @@
+     */
+    protected $runtimeCache = [];
+
+    /**
+     * @Flow\InjectConfiguration(path="statusCode", package="Neos.RedirectHandler")
+     * @var array
+     */
+    protected $defaultStatusCode;
+
+    /**
+     * Returns one redirection for the given $sourceUriPath or NULL if it doesn't exist
+     *
+     * @param string $sourceUriPath
+     * @param string $host Full qualified hostname or host pattern
+     * @return RedirectionDto or NULL if no redirection exists for the given $sourceUriPath
+     * @api
+     */
+    public function getOneBySourceUriPathAndHost($sourceUriPath, $host = null)
+    {
+        $redirection = $this->redirectionRepository->findOneBySourceUriPathAndHost($sourceUriPath, $host);
+        if ($redirection === null) {
+            return null;
+        }
+        return new RedirectionDto($redirection->getSourceUriPath(), $redirection->getTargetUriPath(), $redirection->getStatusCode(), $redirection->getHost());
+    }
+
+    /**
+     * Returns all registered redirection records
+     *
+     * @param string $host Full qualified hostname or host pattern
+     * @return \Generator<RedirectionDto>
+     * @api
+     */
+    public function getAll($host = null)
+    {
+        foreach ($this->redirectionRepository->findAll($host) as $redirection) {
+            yield new RedirectionDto($redirection->getSourceUriPath(), $redirection->getTargetUriPath(), $redirection->getStatusCode(), $redirection->getHost());
+        }
+    }
+
+    /**
+     * Return a list of all host patterns
+     *
+     * @return array
+     */
+    public function getDistinctHosts()
+    {
+        return $this->redirectionRepository->findDistinctHosts();
+    }
+
+    /**
+     * Removes a redirection for the given $sourceUriPath if it exists
+     *
+     * @param string $sourceUriPath
+     * @param string $host Full qualified hostname or host pattern
+     * @return void
+     * @api
+     */
+    public function removeOneBySourceUriPathAndHost($sourceUriPath, $host = null)
+    {
+        $redirection = $this->redirectionRepository->findOneBySourceUriPathAndHost($sourceUriPath, $host);
+        if ($redirection === null) {
+            return;
+        }
+        $this->redirectionRepository->remove($redirection);
+    }
+
+    /**
+     * Removes all registered redirection records
+     *
+     * @param string $host Full qualified hostname or host pattern
+     * @return void
+     * @api
+     */
+    public function removeAll($host = null)
+    {
+        $this->redirectionRepository->removeAll($host);
+    }
+
+    /**
+     * Adds a redirection to the repository and updates related redirection instances accordingly
+     *
+     * @param string $sourceUriPath the relative URI path that should trigger a redirect
+     * @param string $targetUriPath the relative URI path the redirect should point to
+     * @param integer $statusCode the status code of the redirect header
+     * @param array $hosts the list of host patterns
```

Comment:
@aertmann I'm not sure if I understand the question ?

I need to rename host patterns everywhere (because even in Neos we don't support host patterns since sometimes now). So if the question is about host pattern, read "hostname" ;)